app/src/main/java/com/example/plantsapp/presentation/ui/authentication/AuthFragment.kt

```diff
@@ -0,0 +1,63 @@
+package com.example.plantsapp.presentation.ui.authentication
+
+import android.os.Bundle
+import android.view.View
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.commit
+import androidx.fragment.app.viewModels
+import by.kirich1409.viewbindingdelegate.viewBinding
+import com.example.plantsapp.R
+import com.example.plantsapp.databinding.FragmentAuthBinding
+import com.example.plantsapp.presentation.ui.splash.SplashFragment
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class AuthFragment : Fragment(R.layout.fragment_auth) {
+
+    private val binding: FragmentAuthBinding by viewBinding(FragmentAuthBinding::bind)
+    private val viewModel: AuthViewModel by viewModels()
+    private val googleSignInClient by lazy {
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+            .requestIdToken(getString(R.string.default_web_client_id))
+            .requestEmail()
+            .build()
+
+        GoogleSignIn.getClient(requireActivity(), gso)
+    }
+
+    private val googleSignInLauncher =
+        registerForActivityResult(
+            ActivityResultContracts.StartActivityForResult()
+        ) { activityResult ->
+            activityResult.data?.let { viewModel.onSignInResult(it) }
+        }
```

Comment:
Leave TODO here for me to not forget to check this thing in next iteration of pull request